frontend/controller/actions/chatroom.js

```diff
@@ -42,10 +43,10 @@ sbp('okTurtles.events/on', MESSAGE_RECEIVE_RAW, ({
     messageHash: msgData.hash,
     height: msgData.height,
     text: msgData.text,
-    isDMOrMention: isMentionedMe || getters.chatRoomAttributes.type === CHATROOM_TYPES.DIRECT_MESSAGE,
+    isDMOrMention: isMentionedMe || state.attributes?.type === CHATROOM_TYPES.DIRECT_MESSAGE,
     messageType: !newMessage ? MESSAGE_TYPES.TEXT : data.type,
     memberID: innerSigningContractID,
-    chatRoomName: getters.chatRoomAttributes.name
+    chatRoomName: state.attributes?.name
```

Comment:
What is preventing us from moving the getter definitions into a file that is shared by the frontend and the SW?

That would be much preferable to this.

Reply:
Chatroom getters are a Vuex module, which are harder to emulate without Vuex because of state partitioning. In addition, those getters rely on `currentGroupId` and `currentChatroomId` which can't be used in contracts.

Reply:
Is there nothing creative that's possible here? We can't define getters in a way that has them referring to the "the chatroom that corresponds to this `state`"?

Reply:
Thinking about it, another solution along the lines of what you suggested could be something like:

```js
const getters = sbp('chelonia/contract/getters', 'gi.contracts/chatroom', state)
getters.currentChatRoomState
```

But honestly, for simple accessors such as these I don't see the benefit of adding this complexity.

Reply:
@corrideat would it be possible to instantiate dynamically a new batch of getters where `getters. currentChatRoomState` returns a specific contract state in the SW using some sort of factory function?

>  I also think such getters that are simply accessors should be avoided, as they increase code line count and complexity for no benefit.

Getters are very important, as they solve a lot of DRY related problems. By using getters we can ensure that the way that data is accessed everywhere will have consistent return values (e.g. because it uses tricks like `|| {}` etc.). Whereas those considerations would have to be thought of each time you access state data directly.

Reply:
> ```js
> const getters = sbp('chelonia/contract/getters', 'gi.contracts/chatroom', state)
> ```

Yeah, that's a great idea!

I'd make one slight modification: instead of passing in `state`, passing in `contractID`:

```js
const getters = sbp('chelonia/contract/getters', 'gi.contracts/chatroom', contractID)
```

Then yeah, at least we could reuse the contract getters that are already defined in the contracts!

Reply:
Update based on today's conversation: we could use getters _synchronously_ by importing the contract getters from this actions file. However, it'd have the downside of being 'statically hardcoded' based on whatever version is imported at build time and will increase bundle size. Doing it via Chelonia is potentially possible, but it'd need to be _async_ in case the contract isn't cached and need to be loading and also maybe because of sandboxing.

Reply:
Since this no longer seems to be needed in the context of this PR (we're importing getters in the SW as needed), can we make this a new issue, possibly linked to the sandboxing issue (#2364)?